### PR TITLE
Create subject sets from collections

### LINKED ISSUE #297: Import Collections as SubjectSets

### PR DESCRIPTION
**Pull request by edpaget**
Close #297

### CODE REVIEW

File: app/controllers/api/api_controller.rb [hunk 1]
```diff
@@ -14,7 +14,6 @@ class ApiController < ApplicationController
     rescue_from Api::UnauthorizedTokenError,             with: :not_authenticated
     rescue_from Api::UnsupportedMediaType,               with: :unsupported_media_type
     rescue_from ControlControl::AccessDenied,            with: :not_authorized
-    rescue_from ActiveRecord::StatementInvalid,          with: :bad_query
```
Comment (camallen) [inline app/controllers/api/api_controller.rb]: I assume Json Schema Validator is checking id formats now so we don't need to rescue this one explictly?

File: app/controllers/api/v1/subject_sets_controller.rb [hunk 2]
```diff
@@ -5,6 +5,14 @@ class Api::V1::SubjectSetsController < Api::ApiController
...
   schema_type :json_schema
 
-  allowed_params :create
-  allowed_params :update
+  protected
+
+  def build_resource_for_create(create_params)
+    super do |_, link_params|
+      if collection_id = link_params.delete("collection")
+        link_params["subjects"] = Collection.scope_for(:show, api_user)
```
Comment (camallen) [inline app/controllers/api/v1/subject_sets_controller.rb]: Will this return nil or raise and error if the `api_user` has no scope to `view` the collection?

File: app/controllers/api/api_controller.rb (hunk 1 above)
Comment (edpaget) [inline app/controllers/api/api_controller.rb]: The schema is just checking types. The other thing checks if its an
alphabet string or number.

File: app/controllers/api/v1/subject_sets_controller.rb (hunk 2 above)
Comment (edpaget) [inline app/controllers/api/v1/subject_sets_controller.rb]: I'll fix this, it should probably just be following the normal `link_to` code path.

Comment (edpaget) [inline app/controllers/api/v1/subject_sets_controller.rb]: Couldn't figure out a good way to make it use the same code, but it's now a find that will return the same 404 error that a link failure would. (I think #418 changes the specific error code).

**Pull request merged**